### ISSUE TITLE
No keyboard input after compiling

### ISSUE THREAD

**Issue by DowXor** [higan-emu/higan#46]
Hello,

After compiling (with mingw32-make.exe on Windows), everything works except the keyboard input: in options as in games.

Can you help me ?

Thanks

**Comment by Screwtapello** [comment]
Are you using higan or byuu?

In higan, in the panel at the bottom of the window there's a drop-down list on the left. If you choose "Settings". In the list below the drop-down, choose "Input" (with the icon of a joystick). In the pane to the right, make sure "Driver" is set to "Windows", then click "Change" to make sure the driver is active.

In byuu, from the Settings menu choose Drivers. Underneath "Input" heading, "Driver" should be set to "Windows", then click "Reload" to make sure that driver is active.

Once that's done, you can try setting some hotkeys (like "toggle fullscreen") to be sure the emulator can read keypresses. Once *that's* done, you can try connecting an emulated controller to the emulated console and setting keybindings for that.

**Comment by Screwtapello** [comment]
I don't know if commenting out that line should cause exactly that problem, but I'm pretty sure that commenting it out would cause a bunch of problems, so it doesn't surprise me.

What version/distribution of mingw32 are you using? `clock_gettime()` is technically a POSIX function, so it wouldn't normally be present on Windows, but higan's makefile [specifically adds](https://github.com/higan-emu/higan/blob/master/nall/GNUmakefile#L116) the `-lpthreads` compiler flag on Windows, which should provide the required implementation (at least, it does for the nightly builds).

**Comment by DowXor** [comment]
Yeah, I suspected that if I comment on this line, it could generate errors: D

I am using version 9.2.0 (x64). I downloaded it here: https://jmeubank.github.io/tdm-gcc/

![image](https://user-images.githubusercontent.com/57010998/79840761-a85a5400-83b6-11ea-9c7a-3040a55962bd.png)

![image](https://user-images.githubusercontent.com/57010998/79840775-aee8cb80-83b6-11ea-8ce6-7dcc43bb1399.png)

**Comment by Screwtapello** [comment]
bsnes and higan always used to be built with TDM-GCC because it was nice and self-contained, but after it went on hiatus I think byuu started using the full MSYS environment to get an updated version of GCC.

Apparently the `clock_gettime` function should be declared in a file called `pthread_time.h`. If you can find your copy, take a look inside. It should look [like this](https://github.com/Alexpux/mingw-w64/blob/d0d7f784833bbb0b2d279310ddc6afb52fe47a46/mingw-w64-libraries/winpthreads/include/pthread_time.h), but if it just says:

    /* Dummy header, which gets overriden, if winpthread library gets installed.  */

...then I suspect that version of TDM-GCC isn't including the winpthread library. [The README](https://github.com/jmeubank/tdm-distrib/blob/master/tdm64/core/README-gcc-tdm64.md) certainly suggests it should be included, but... I have no idea.

**Comment by DowXor** [comment]
Ok, thanks for your this. I will do tests and I will tell you if it works.
